text/0006-equivocation-catastrophes.md

```diff
@@ -0,0 +1,114 @@
+
+In order to make sure that as many validators as possible become aware of all the equivocations eventually, the operation of the nodes shouldn't stop. Finalizing blocks should be halted, as finalization ceases to mean anything when it can be reverted, but the nodes should continue gossiping votes.
+
+The nodes could also output the hash of the last block finalized before FTT has been exceeded and all the transactions included in the later blocks, and the list of equivocators. These will be useful for the manual steps.
+
+The manual actions afterwards will consist of two main steps:
+
+1. Reaching social consensus about the new starting point,
+2. Restarting the network with the new starting point.
+
+We will go into more detail about these steps below.
+
+### Social consensus
+
+For the initial version of this CEP, we assume that we have some kind of platform that enables open communication between validators. The chosen platform should allow for storing, linking, referencing and displaying the whole public process and communication. We also should make sure that the content is in a format that can be displayed elsewhere.
+
+The first essential step will be determining the starting point for the network after the restart. The suggested approach here would be to agree upon the new initial global state, new set of validators and restart the network as if it was starting from scratch.
+
+A desirable goal for the initial global state would be to revert as few transactions from all the forks as possible, in order to limit the damage. Every transaction that was included in a block which appeared to be finalized for a time could have caused real world consequences (like, for example, a merchant sending ordered goods). Reverting any such transaction might potentially cause a loss to a user.
+
+In order to minimize losses, the following approach is proposed: first, choose the last safe block as the starting point. A safe block would be one that still appears finalized with some small FTT, for example 1%, after a full synchronization of the protocol state.
+
+After the block is chosen, take its global state and all transactions from the blocks that were considered finalized at some point, but are no longer a part of the main branch of the chain, and attempt to replay them on top of that global state. Since the order could matter, the community would have to agree on one (the default one could be based on deploy timestamps, but the community could decide on some other ordering if deemed necessary). The deploys' TTL could also be expanded to account for the time spent reaching social consensus and allow them to still be valid afterwards. Call global state resulting from replaying these deploys the "pre-initial state".
```

Comment:
> The deploys' TTL could also be expanded to account for the time spent reaching social consensus and allow them to still be valid afterwards. Call global state resulting from replaying these deploys the "pre-initial state".

Since deploy's TTL is also signed over, this requires changes to the node software that filters out expired deploys.